From 089057d3c574ec804c967dcd4ae8a0368e7ee90f Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Thu, 23 Jun 2016 14:41:12 -0400
Subject: [PATCH 55/55] Change debug entry insertion order (rhbz#1285601)

Customer reports indicated that it was confusing to have the debugging
entries before non-debugging entries. Altered the insertion code to use
the --set-index feature on the debug kernels to force them to insert
after the initial non-debugging entry.

Resolves: rhbz#1285601
---
 new-kernel-pkg | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/new-kernel-pkg b/new-kernel-pkg
index bd28634..847e959 100755
--- a/new-kernel-pkg
+++ b/new-kernel-pkg
@@ -238,7 +238,7 @@ install() {
 		--remove-kernel=\"TITLE=$title\$debugtitle\""
 
 	rungrubby ${ARGS} $makedefault
-	rungrubby --debug ${ARGS}
+	rungrubby --debug ${ARGS} --set-index=1
     else
 	[ -n "$verbose" ] && echo "$grubConfig does not exist, not running grubby for grub 0.97"
     fi
@@ -252,7 +252,7 @@ install() {
 		--args=\"root=$rootdevice $kernargs \$debugargs\" \
 		--remove-kernel=\"TITLE=$title\$debugtitle\""
 	rungrubby ${ARGS} $makedefault
-	rungrubby --debug ${ARGS}
+	rungrubby --debug ${ARGS} --set-index=1
     else
 	[ -n "$verbose" ] && echo "$grub2Config does not exist, not running grubby for grub 2"
     fi
@@ -267,7 +267,7 @@ install() {
 		--args=\"root=$rootdevice $kernargs \$debugargs\" \
 		--remove-kernel=\"TITLE=$title\$debugtitle\""
 	rungrubby ${ARGS} $makedefault
-	rungrubby --debug ${ARGS}
+	rungrubby --debug ${ARGS} --set-index=1
     else
 	[ -n "$verbose" ] && echo "$grub2EfiConfig does not exist, not running grubby for grub 2 with UEFI"
     fi
@@ -283,7 +283,7 @@ install() {
 		--remove-kernel=\"TITLE=$version\""
 
 	rungrubby ${ARGS} $makedefault
-	rungrubby --debug ${ARGS}
+	rungrubby --debug ${ARGS} --set-index=1
 	if [ -n "$runLilo" ]; then
 	    [ -n "$verbose" ] && echo "running $lilo"
 	    if [ ! -x $lilo ] ; then
@@ -307,7 +307,7 @@ install() {
 	    --remove-kernel=\"TITLE=$title\$debugtitle\""
 
 	rungrubby ${ARGS} $makedefault
-	rungrubby --debug ${ARGS}
+	rungrubby --debug ${ARGS} --set-index=1
     else
 	[ -n "$verbose" ] && echo "$extlinuxConfig does not exist, not running grubby for extlinux"
     fi
-- 
2.5.5

